ui-react/packages/atlasmap/src/Views/NamespaceTableView.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FunctionComponent, ReactElement } from "react";
+
+import {
+  Bullseye,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Title,
+  Level,
+  LevelItem,
+  Tooltip,
+  Button,
+} from "@patternfly/react-core";
+import { TableIcon, PlusIcon } from "@patternfly/react-icons";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  IRowData,
+  textCenter,
+  ICell,
+} from "@patternfly/react-table";
+
+import { MainContent } from "../Layout";
+import { IAtlasmapDocument } from ".";
+import { Actions } from "../UI";
+import { ISourceColumnCallbacks } from "./ColumnMapperView";
+
+const styles = StyleSheet.create({
+  title: { marginTop: "1rem", marginBottom: "1rem" },
+});
+
+export interface INamespaceTableProps {
+  sources: IAtlasmapDocument[];
+  sourceEvents: ISourceColumnCallbacks;
+}
+
+export const NamespaceTableView: FunctionComponent<INamespaceTableProps> = ({
+  sources,
+  sourceEvents,
+}) => {
+  const columns = [
+    "Alias",
+    "URI",
+    "Location URI",
+    {
+      title: "Target Namespace",
+      transforms: [textCenter],
+      cellTransforms: [textCenter],
+    },
+  ];
+
+  return sources.length === 0 ? (
+    <Bullseye>
+      <EmptyState variant={EmptyStateVariant.small}>
+        <EmptyStateIcon icon={TableIcon} />
+        <Title headingLevel="h2" size="lg">
+          No sources found
+        </Title>
+      </EmptyState>
+    </Bullseye>
+  ) : (
+    <MainContent>
+      {sources.reduce(
```

Comment:
I’d move this block before the return, it’s pretty complex and difficult to read inside the jsx